src/AWS.Deploy.Common/Recipes/Validation/IOptionSettingItemValidator.cs

```diff
@@ -10,6 +10,12 @@ namespace AWS.Deploy.Common.Recipes.Validation
     /// </summary>
     public interface IOptionSettingItemValidator
     {
-        ValidationResult Validate(object input);
+        /// <summary>
+        /// Validates an override value for an OptionSetting
+        /// </summary>
+        /// <param name="input">Raw input for the current OptionSetting</param>
+        /// <param name="recommendation">Current deployment recommendation, may be used if the validator needs to consider properties other than itself</param>
+        /// <returns>Whether or not the input is valid</returns>
+        ValidationResult Validate(object input, Recommendation recommendation);
```

Comment:
Left as-is for now - prefer to require it everywhere since we're going to have to always pass it in anyway from the actual source since we're generally looping over `IOptionSettingItemValidator[]`.

Would rather have a compile-time error rather than a runtime error when a validator asserts that it was passed in. 

Though a lot has shifted in a month and I know you're working on option dependencies, so let me know if you'd still like to see changes.